biogeophys/EDSurfaceAlbedoMod.F90

```diff
@@ -195,756 +125,918 @@ subroutine ED_Norman_Radiation (nsites, sites, bc_in, bc_out )
+    real(r8) :: gdir
+    
+
+    real(r8), parameter :: forc_dir(n_rad_stream_types) = (/ 1.0_r8, 0.0_r8 /)   ! These are binary switches used
+    real(r8), parameter :: forc_dif(n_rad_stream_types) = (/ 0.0_r8, 1.0_r8 /)   ! to turn off and on radiation streams
+
+
+    
+    associate(&
+         rhol         =>    EDPftvarcon_inst%rhol                     , & ! Input:  [real(r8) (:)   ] leaf reflectance: 1=vis, 2=nir
+         rhos         =>    EDPftvarcon_inst%rhos                     , & ! Input:  [real(r8) (:)   ] stem reflectance: 1=vis, 2=nir
+         taul         =>    EDPftvarcon_inst%taul                     , & ! Input:  [real(r8) (:)   ] leaf transmittance: 1=vis, 2=nir
+         taus         =>    EDPftvarcon_inst%taus                     , & ! Input:  [real(r8) (:)   ] stem transmittance: 1=vis, 2=nir
+         xl           =>    EDPftvarcon_inst%xl                       , & ! Input:  [real(r8) (:)   ] ecophys const - leaf/stem orientation index
+         clumping_index  => EDPftvarcon_inst%clumping_index) 
+
+
+      ! Initialize local arrays
+
+      weighted_dir_tr(:)   = 0._r8
+      weighted_dif_down(:) = 0._r8
+      weighted_dif_up(:)   = 0._r8
+
+      ftweight(:,:,:)      = 0._r8
+      tr_dir_z(:,:,:)      = 0._r8   
+      tr_dif_z(:,:,:)      = 0._r8
+      lai_change(:,:,:)    = 0._r8
+      Dif_up(:,:,:)        = 0._r8
+      Dif_dn(:,:,:)        = 0._r8
+      refl_dif(:,:,:,:)    = 0.0_r8
+      tran_dif(:,:,:,:)    = 0.0_r8
+      dif_ratio(:,:,:,:)   = 0.0_r8
+
+
+      ! Initialize the ouput arrays
+      ! ---------------------------------------------------------------------------------
+      albd_parb_out(1:hlm_numSWb) = 0.0_r8
+      albi_parb_out(1:hlm_numSWb) = 0.0_r8
+      fabd_parb_out(1:hlm_numSWb) = 0.0_r8
+      fabi_parb_out(1:hlm_numSWb) = 0.0_r8
+      ftdd_parb_out(1:hlm_numSWb) = 1.0_r8
+      ftid_parb_out(1:hlm_numSWb) = 1.0_r8
+      ftii_parb_out(1:hlm_numSWb) = 1.0_r8
+                 
+      ! Is this pft/canopy layer combination present in this patch?
+      
+      ! IS THIS NEEDED? SHOULDN'T THIS ALREADY BE CALCULATED??? RGK
+
+      do L = 1,nclmax
+         do ft = 1,numpft
+            currentPatch%canopy_mask(L,ft) = 0
+            do  iv = 1, currentPatch%nrad(L,ft)
+               if (currentPatch%canopy_area_profile(L,ft,iv) > 0._r8)then
+                  currentPatch%canopy_mask(L,ft) = 1
+                  !I think 'present' is only used here...
+               endif
+            end do !iv
+         end do !ft
+      end do !L
+
+
+      !++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+      ! Direct beam extinction coefficient, k_dir. PFT specific.
+      !++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+      cosz = max(0.001_r8, currentPatch%solar_zenith_angle ) !copied from previous radiation code...
+      do ft = 1,numpft
+         sb = (90._r8 - (acos(cosz)*180/pi)) * (pi / 180._r8)
```

Comment:
```suggestion
         sb = (90._r8 - (acos(cosz)*180._r8/pi)) * (pi / 180._r8)
```

Reply:
got that change in my current working version, I'm currently working through a bug so will issue more changes.